src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -126,6 +126,17 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
                            .Cast<IVsReferenceItem>()
                            .Select(ToPackageDownloadDependency));
                 }
+
+                if(targetFrameworkInfo2.FrameworkReferences != null)
+                {
+                    var uniqueFrameworkRefs = new HashSet<string>(
+                        targetFrameworkInfo2.FrameworkReferences
+                            .Cast<IVsReferenceItem>()
+                            .Select(e => e.Name),
+                        StringComparer.OrdinalIgnoreCase);
+
+                    tfi.FrameworkReferences.AddRange(uniqueFrameworkRefs);
```

Comment:
Can you just make `FrameworkReferences` a hash set, since order doesn't matter and skip the intermediate hash set?

Reply:
yup, made them a hash set.